Sources/Private/MainThread/LayerContainers/Utility/CachedImageProvider.swift

```diff
@@ -21,7 +21,7 @@ private final class CachedImageProvider: AnimationImageProvider {
   // MARK: Public
 
   public func imageForAsset(asset: ImageAsset) -> CGImage? {
-    if let image = imageCache.object(forKey: asset.id as NSString) {
+    if imageProvider.cacheEligible, let image = imageCache.object(forKey: asset.id as NSString) {
```

Comment:
Right now this is unnecessarily caching images in the `imageCache` which are never used.

Instead of creating a `CachedImageProvider` but just not using it, perhaps instead we should update the `var cachedImageProvider` implementation below:

```swift
extension AnimationImageProvider {
  /// Create a cache enabled image provider which will reuse the asset image with the same asset id
  /// It wraps the current provider as image loader, and uses `NSCache` to cache the images for resue.
  /// The cache will be reset when the `animation` is reset.
  /// If this image provider is not eligible for caching, simply returns self.
  var cachedImageProvider: AnimationImageProvider {
    guard cacheEligibile else { return self }
    return CachedImageProvider(imageProvider: self)
  }
}
```

Reply:
But I think other built-in providers, like FilepathImageProvider. Having a memory cache is necessary, and we should leave the choice to the dev user to handle some special scenarios.

Reply:
Not sure what you mean about `FilepathImageProvider`. In that case `imageProvider.cacheEligible` will be `true` so there will be no behavior change no matter what.

I think we shouldn't be writing images to the cache in the case where `imageProvider.cacheEligible` is `false`. Those cached images will not be used, but writing them to the cache consumes additional memory.

Reply:
Yeah,I looked at the source code carefully, and indeed if cacheEligibile is not true , Only return our Custom AnimationImageProvider it's ok, your solution is feasible.